api/api_sources/sources/database/models/controllers/user.controller.ts

```diff
@@ -64,6 +69,40 @@ export class UserDataController extends DataModelController<User> {
         }
         return;
     }
+
+    /**
+     * @description Returns true if there are pending requests matching the given roleId
+     * @param allRequest all pending requests
+     * @param roleCodeId role id
+     * @returns boolean
+     */
+    public getPendingRequestForRole(allRequest: RequestAccess[], roleCodeId: number): boolean {
+        if (!allRequest || allRequest.length === 0) {
+            return false;
+        }
+
+        return (allRequest.filter(request => request.requestedAccessCode.role_code_id === roleCodeId).length > 0);
+    }
+
+    /**
+     * @description Process all the pending requests and return the status associated with each role
+     * @param allRequest all pending requests
+     * @returns pending status for different roles
+     */
+    public async getPendingStatus(allRequest: RequestAccess[]): Promise<any> {
```

Comment:
The method is an async type but there is no line of code that handling promise or using await. So, please review it.

Reply:
Good catch, I forgot to remove it